frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx

```diff
@@ -323,20 +326,30 @@ export class UnconnectedDataSelector extends Component {
 
   async componentDidMount() {
     const { activeStep } = this.state;
-    const sourceId = this.props.selectedTableId;
+    const {
+      fetchFields,
+      fetchQuestion,
+      selectedDataBucketId,
+      selectedQuestion,
+      selectedTableId: sourceId,
+    } = this.props;
 
     if (!this.isLoadingDatasets() && !activeStep) {
       await this.hydrateActiveStep();
     }
 
-    if (this.props.selectedDataBucketId === DATA_BUCKET.MODELS) {
-      this.showSavedEntityPicker();
+    if (selectedDataBucketId === DATA_BUCKET.MODELS) {
+      this.showSavedEntityPicker({ entityType: "model" });
     }
 
     if (sourceId) {
-      await this.props.fetchFields(sourceId);
+      await fetchFields(sourceId);
       if (this.isSavedEntitySelected()) {
-        this.showSavedEntityPicker();
+        fetchQuestion(sourceId);
+
+        this.showSavedEntityPicker({
+          entityType: selectedQuestion?.type(),
```

Comment:
AFAIK this won't work if you don't use `this.props.selectedQuestion` here, otherwise you would reference the `undefined` value before the `fetchQuestion` call. 

Btw, `fetchQuestion` should be await'ed as well.